tests/shadowcasting_test.cpp

```diff
@@ -443,6 +431,83 @@ void shadowcasting_3d_2d( int iterations )
     REQUIRE( passed );
 }
 
+
+// T, O and V are 'T'ransparent, 'O'paque and 'V'isible.
+// X marks the player location, which is not set to visible by this algorithm.
+#define T LIGHT_TRANSPARENCY_CLEAR
+#define O LIGHT_TRANSPARENCY_SOLID
+#define V LIGHT_TRANSPARENCY_CLEAR
+#define X LIGHT_TRANSPARENCY_SOLID
+
+TEST_CASE( "shadowcasting_spot_checks" ) {
+    float seen_squares[MAPSIZE*SEEX][MAPSIZE*SEEY] = {{ 0 }};
+    float transparency_cache[MAPSIZE*SEEX][MAPSIZE*SEEY] = {{ 0 }};
```

Comment:
Here you use the defines as if the caches were addressed by x and then by y, but below you address them by y first.
If `SEEX != SEEY`, this would break.